`check_outliers` paper

rempsyc opened the pull request:
Draft of the `check_outliers` paper (re: https://github.com/orgs/easystats/teams/core-team/discussions/5).

Looking forward to any and all feedback and contributions @easystats/core-team!

PS: reminder to use [skip ci] in your pushes (don't keep forgetting like me!!).

rempsyc commented:
This is the link to the PDF :https://github.com/easystats/performance/blob/fcec6306987475b250ffb063b1c8e6e2ba3bbf48/papers/Mathematics/paper.pdf

This is the link to the rmarkdown file (the one to edit for any changes or comments): https://github.com/easystats/performance/blob/fcec6306987475b250ffb063b1c8e6e2ba3bbf48/papers/Mathematics/paper.Rmd

mattansb commented:
I've added some edits to reflect my own POV on the topic - specifically that outlier detection tools are only part of the picture, a tool to be used by humans to "flag" suspect outliers, but that the final decision should align with domain knowledge. Happy to smooth those points out some more (:

I also know very little about the maths of these methods - but did you know you can just select-copy formulas from Wikipedia and they paste as latex??

rempsyc commented:
@easystats/core-team I have integrated the earlier comments for the outliers paper. A reminder that there is little over 2 weeks before the submission deadline, so now is a great time for contributions for anyone who hasn't contributed yet. I would like to keep the last week before submission to finalize all the last details that were not already addressed yet (e.g., writing the abstract if no one else did).

Here's a few "easy" things it is still possible to contribute to (beyond expertise-based feedback of course):
1. Writing the abstract
2. I feel like one weakness of the paper is that we rely almost exclusively on the two Leys/Lakens papers. Perhaps adding a bit of citation variety would make the paper stronger.
3. Chip in or whether to keep or remove formulas
4. Flesh out a paragraph based on @mattansb's comment on model-based outliers below:

> Something something... what is leverage... why we should care if few observations have (relatively) strong leverage (answer: they are suspect of biasing our estimates!).

rempsyc commented:
@bwiernik , it would also be great if you could go over the paper if you have the time. Here's a few lingering stats questions beyond my competence level for you:

1. I know you prefer model-based outlier detection methods over multivariate methods (and I now do as well). Do we have any references to back this or is it just common sense/opinion? How would you write a justification for this in the paper?
2. I write that the model-based influence plot represents the Cook's distance. Is that wrong? Is it just leverage/influential observations? Then how should I describe it in the caption? What is the relationship to Cook’s distance if that was the selected method?
3. For z-score methods, what's our justification for using a threshold of 3.09 instead of 3 like suggested by Leys et al. (2019)? Anything we can cite to support this discrepancy or justify it? Or should we change it to align with recommandations?
4. Leys et al. (2019) write,

> MCD75 (breakdown point = 0.25) (or the MCD50 (breakdown point = 0.5) if you suspect the presence of more than 25% of outlying values)".

We use a threshold of `stats::qchisq(p = 1 - 0.001, df = ncol(x))` (18.467 in the example in the paper). So how do these thresholds compare, and should we align our threshold with this recommendation, or else how to justify it?

bwiernik commented:
I'll take a look tonight

bwiernik commented:
For 3, it's the critical value for p < .001 (ie, `qnorm(.999)`. That is consistent with all of our other thresholds. That also seems to be what the "3" in Leys et al is an approximation of. I suggest being consistent across indices and using the critical value for .999 (so 3.09023…)

rempsyc commented:
For 1, seems like Leys (the people we keep citing) actually explicitly recommends robust Mahalanobis (MCD) over Cook's distance in one of their papers I have missed:

>### Cook's distance and leverage methods
> Among outlier detection methods, Cook's distance and leverage are less common than the basic Mahalanobis distance, but still used. Cook's distance estimates the variations in regression coefficients after removing each observation, one by one (Cook, 1977). Therefore, as soon as there is more than one outlying value, the remaining outliers influence the estimators. As for the leverage method, it provides the same information as the Mahalanobis distance (Cohen et al., 2003): It is based on the study of residuals and their distance from the mean vector (e.g. Thode, 2002), which are computed using mean and variance, still polluted by outliers. **_This is why we recommend to use robust procedures to estimate the position μ and the scatter matrix Σ._** The aim of robust methods is to estimate the location μ and the scatter matrix Σ even though the data has been contaminated. We introduce in the remaining of the text the robust method called Minimum Covariance Determinant.

https://doi.org/10.1016/j.jesp.2017.09.011

So... do we have robust model-based outlier detection methods? In the paper I was going to suggest to always prioritize model-based methods, but with the above, what does it change?

The paper below, based on a simulation comparing Cook to Mahalanobis, also does not recommend Cook:

http://article.sapub.org/10.5923.j.ajms.20150501.06.html

bwiernik commented:
I pretty strongly disagree with them. If there are a bunch of outliers you're concerned about then you should use a robust regression method. The idea of results-agnostic outlier evaluation is pretty silly to me. 

The overarching philosophy I think is good to communicate is that unusual events happen, and we shouldn't arbitrarily discard them. We should expect to see some "outliers" in a decently sized sample. Approaches like Mahalanobis D + heuristic tend to flag too many "outliers". If you are concerned about outliers, use a robust model like T regression or quantile regression.

mattansb commented:
💯% agree with @bwiernik (which was also the point I was trying to get across in the suggestions/edits I made).

rempsyc commented:
Ok. Good. Very good actually because this disagreement could make for an even more interesting, meaningful, and unique point of contribution.

mattansb commented:
![LetThemGIF](https://user-images.githubusercontent.com/35330040/218042400-c8de18dc-cb77-4e5f-89d4-019b4d5dcfa2.gif)

rempsyc commented:
Still from Leys et al. (2018), in the intro, they're actually calling Cook and leverage (so indirectly _easystats_) questionnable methods 🤔

> A survey made in the same journals as those used by Leys et al. (2013), namely the Journal of Personality and Social Psychology (JPSP) and Psychological Science (PS), revealed that few researchers seem to mind about multivariate outliers. [...] From these 24 papers, nine used the basic Mahalanobis distance, **_five used another criterion (leverage using Student-t residuals or Cook's distance)_**, and ten did not provide any information about the detection strategy. [...] This means that for over 97.5% of this type of multivariate analyses, either researchers did not search for multivariate outliers or they did not report any information about it. The 16 [15*?] other teams looked for multivariate outliers, but either **_with a questionable method_** or without providing information about the method.

Ok, the war is on

![40ebafd4cdb31c02a4169b63a2004ae4](https://user-images.githubusercontent.com/13123390/218278742-de3cddf7-8c73-471b-b4f4-cdc94b3bf381.gif)

rempsyc commented:
Ok, I have made some new changes to the paper. I am by no means an expert on statistics and even more so on statistical outlier detection methods. So your feedback is important to make sure I'm not inadvertently writing bs :-)

@easystats/core-team this is also the last week to make contributions to the paper, as next weekend I would like to wrap it up so we have a few days to review the final version before submission. Thanks!

DominiqueMakowski commented:
Good job so far @rempsyc ! I've done some edits to the first half and will continue tomorrow the second half but in the meantime feel free to overwrite :)

DominiqueMakowski commented:
![image](https://user-images.githubusercontent.com/8875533/218338396-709fc244-a20f-426a-909c-9f7070b1e78b.png)

Also, was there a particular rationale behind plotting the normalized distance? Wouldnt it be more informative to have the raw distance, which values are often quite informative (especially in the univariate case)?

rempsyc commented:
Thanks for your changes @DominiqueMakowski! Looking forward to tomorrow changes as well :)

> Also, was there a particular rationale behind plotting the normalized distance? Wouldnt it be more informative to have the raw distance, which values are often quite informative (especially in the univariate case)?

That was the goal in easystats/see#262. However, I don't think it will be resolved before the submission deadline unfortunately. I'd very much like to plot the raw values as well. But the rationale so far was that this is the default plotting method for a `check_outliers` object, even for univariate cases.

Unless you tell me there is a secret argument we can feed to the generic `plot()` function to get the raw values, we would need to go dig them from the attributes. And we could do it, but don't think it is very user-friendly, especially for this paper where we're trying to show them how simple it is.

If we think this should be the default, we should change the default so that when a single method is selected, the distance is in the original scale instead of normalized. But I don't know if this will happen before submission.

DominiqueMakowski commented:
Ah I see, perhaps we'll have a chance to update the figure in between the revision 🤞 

> Edit: + reminder to use [skip ci] 

oops, I just say I completely missed most of last months' discussions, including the whole infrastructure talks... but shhh or I'm pretty sure indra will create a special GH action to kill me

<sub>[skip ci]</sub>

rempsyc commented:
> For 3, it's the critical value for p < .001 (ie, qnorm(.999). That is consistent with all of our other thresholds. That also seems to be what the "3" in Leys et al is an approximation of. I suggest being consistent across indices and using the critical value for .999 (so 3.09023…)

But should it not be 3.29 if it's a two-tailed p value of .001 (so .0005 each side of the distribution)? Unless we use one-tailed p values?

mattansb commented:
I am also familiar with 3.29...

rempsyc commented:
@strengejacke I think you're the only one missing from the paper at this point. Any chance you might give this a look, make a few changes, and add yourself as author? :D

strengejacke commented:
Yes, I can look at the paper see if I can make any "substantial" contributions to it. My question is, which of the current paper drafts is considered for publication in a "higher impact journal"? (by that I mean listed in pubmed and some kind of IF).

rempsyc commented:
Both this `check_outliers` draft and Mattan's fei effect size draft are for the special issue "Advances in Statistical Computing" in the journal Mathematic (IF: 2.592).

rempsyc commented:
@easystats/core-team, thank you for your work on this paper so far. I will be submitting it in a couple of days max.

I have now finalized the paper, so this is the last opportunity to make any last-minute changes. Please reread the formatted PDF version. The latest PDF is available here: https://github.com/rempsyc/performance/blob/check_outliers_paper/papers/Mathematics/paper.pdf

**Please check that your name and affiliation are correct**. You also **must** approve of the final version, by simply upvoting this post with a 👍

FYI, in this version, I have:
- Written the abstract
- Removed the maths formulas
- Updated the default threshold for _z_ score methods to 3.29 (thanks @mattansb for confirming this)

rempsyc commented:
@IndrajeetPatil @mattansb please check that the conclusion jibes with you this time :P

@bwiernik would you like to add yourself as author including your affiliations (e.g., independent researcher) after rereading the paper?

bwiernik commented on papers/Mathematics/paper.Rmd:
```diff
@@ -0,0 +1,264 @@
+  - name: Indrajeet Patil
+    affil: 3
+    orcid: 0000-0003-1995-6531
+  - name: Dominique Makowski
+    affil: 4
+    orcid: 0000-0001-5375-9967
```
```suggestion
  - name: Dominique Makowski
    affil: 4
    orcid: 0000-0001-5375-9967
  - name: Brenton M. Wiernik
    affil: 5
    orcid: 0000-0001-9560-6336
```

bwiernik commented on papers/Mathematics/paper.Rmd:
```diff
@@ -0,0 +1,264 @@
+    address: Independent Researcher
+  - num: 3
+    address: Center for Humans and Machines, Max Planck Institute for Human Development, Berlin, Germany 
+  - num: 4
+    address: School of Psychology, University of Sussex, Brighton, UK
```
```suggestion
  - num: 4
    address: School of Psychology, University of Sussex, Brighton, UK
  - num: 5
    address: Independent Researcher, Tampa, FL, USA
```

bwiernik commented:
Looks good to me! Great work

rempsyc commented:
@DominiqueMakowski in your original post you said,

> The deadline is the 23rd February 2023

But on this page I see: https://www.mdpi.com/journal/mathematics/special_issues/Advances_in_Statistical_Computing

> Deadline for manuscript submissions: 20 February 2023 | Viewed by 1546

So deadline is tomorrow actually? Doesn't give much time for all to approve the manuscript... So which date is correct?

strengejacke commented:
> @easystats/core-team, thank you for your work on this paper so far. I will be submitting it in a couple of days max.
> 
> I have now finalized the paper, so this is the last opportunity to make any last-minute changes. Please reread the formatted PDF version. The latest PDF is available here: https://github.com/rempsyc/performance/blob/check_outliers_paper/papers/Mathematics/paper.pdf
> 
> **Please check that your name and affiliation are correct**. You also **must** approve of the final version, by simply upvoting this post with a 👍
> 
> FYI, in this version, I have:
> 
> * Written the abstract
> * Removed the maths formulas
> * Updated the default threshold for _z_ score methods to 3.29 (thanks @mattansb for confirming this)

I'll use today to read the paper and add some editions/comments, when necessary.

strengejacke commented:
@DominiqueMakowski initially mentioned the 23rd February. Website states 20th... ?

DominiqueMakowski commented:
Website states 20th indeed... not sure what happened, but anyways, we are asking for an extension. Will keep you updated but it should be alright.

DominiqueMakowski commented on papers/Mathematics/paper.Rmd:
```diff
@@ -13,6 +13,9 @@ author:
   - name: Dominique Makowski
     affil: 4
     orcid: 0000-0001-5375-9967
+  - name: Brenton M. Wiernik
```
Also if nobody opposes (@bwiernik @rempsyc) I wouldn't mind being last here... the academic path won't grind itself 🥲

bwiernik replied:
No preferences from me

rempsyc replied:
Sure, feel free to change it

IndrajeetPatil replied:
Go for it!

strengejacke commented:
@rempsyc nice paper! As you can see, I have some comments, which you may consider or not :-)

mattansb commented:
Looks awesome @rempsyc !

strengejacke commented:
Sorry, I didn't submit my comments ^^

rempsyc commented:
Thanks for your comments @strengejacke, those are great and very useful! Fortunately you thought of pressing that submit button in time 🙃

rempsyc commented on papers/Mathematics/paper.Rmd:
```diff
@@ -0,0 +1,269 @@
+
+Nonetheless, the improper handling of these outliers can substantially affect statistical model estimations, biasing effect estimations and weakening the models' predictive performance.
+It is thus essential to address this problem in a thoughtful manner. Despite the existence of established recommendations and guidelines, many researchers still do not treat outliers in a consistent manner, or do so using inappropriate strategies [@simmons2011false; @leys2013outliers].
+Fortunately, guidelines exist in this regard. Yet, especially in the field of psychology, many researchers still do not treat outliers in a consistent manner or do so using inappropriate strategies [@simmons2011false; @leys2013outliers].
```
✔

rempsyc commented on papers/Mathematics/paper.Rmd:
```diff
@@ -0,0 +1,269 @@
+Real-life data often contain observations that can be considered *abnormal* when compared to the main population. The cause of it---be it because they belong to a different distribution (originating from a different generative process) or simply being extreme cases, statistically rare but not impossible---can be hard to assess, and the boundaries of "abnormal" are hard to define.
+
+Nonetheless, the improper handling of these outliers can substantially affect statistical model estimations, biasing effect estimations and weakening the models' predictive performance.
+It is thus essential to address this problem in a thoughtful manner. Despite the existence of established recommendations and guidelines, many researchers still do not treat outliers in a consistent manner, or do so using inappropriate strategies [@simmons2011false; @leys2013outliers].
```
I included it because of this:

> In a recent article, Simmons, Nelson, and Simonsohn (2011) showed how, due to the misuse of statistical tools, significant results could easily turn out to be false positives (i.e., effects considered significant whereas the null hypothesis is actually true). In their argument, they accurately pinpointed the importance of outliers.

From the Simmons paper: 

> Ambiguity is rampant in empirical research. As an example, consider a very simple decision faced by researchers analyzing reaction times: how to treat outliers. In a perusal of roughly 30 Psychological Science articles, we discovered considerable inconsistency in, and hence considerable ambiguity about, this decision.

rempsyc commented on papers/Mathematics/paper.Rmd:
```diff
@@ -0,0 +1,269 @@
+
+Nonetheless, which exact outlier method to use depends on many factors. In some cases, eye-gauging odd observations can be an appropriate solution, though many researchers will favour algorithmic solutions to detect potential outliers, for example, based on a mathematical score.
+
+One of the factor to consider when selecting an algorithmic outlier detection method is the statistical test of interest. When using a regression model, for example, relevant information can be found by identifying observations that do not fit well with the model. This approach, known as model-based outliers detection (as outliers are extracted after the statistical model has been fit), can be co ...
```
The way I see it is that the sentence about the test of interest introduces the next paragraph as well. So first paragraph is model-based tests, next paragraph we talk about SEM, _t_ tests, and correlations.

rempsyc commented on papers/Mathematics/paper.Rmd:
```diff
@@ -0,0 +1,269 @@
+Although many researchers attempt to identify outliers with measures based on the mean (e.g., _z_ scores), those methods are problematic because the mean and standard deviation themselves are not robust to the influence of outliers and they assume a normal distribution. Therefore, current guidelines recommend using robust methods to identify outliers, such as those relying on the median as oppose ...
+
+Nonetheless, which exact outlier method to use depends on many factors. In some cases, eye-gauging odd observations can be an appropriate solution, though many researchers will favour algorithmic solutions to detect potential outliers, for example, based on a mathematical score.
+
```
@DominiqueMakowski added this sentence. Dom any reference about many researchers favouring algorithmic solutions to detect outliers?

rempsyc replied:
Bump @DominiqueMakowski

rempsyc commented on papers/Mathematics/paper.Rmd:
````diff
@@ -0,0 +1,269 @@
+plot(outliers)
+```
+
+Other univariate methods are available, such as using the interquartile range (IQR), or based on different intervals, such as the Highest Density Interval (HDI) or the Bias Corrected and Accelerated Interval (BCI). These methods are documented and described in the function's [help page](<https://easystats.github.io/performance/reference/check_outliers.html>).
````
I have no idea about those, that's why I try to stick to recommandations 😅. @DominiqueMakowski added this part. Dom could you explain why/when it might be useful to use an interval instead of, say, MAD or IQR?

rempsyc replied:
Bump @DominiqueMakowski

rempsyc commented on papers/Mathematics/paper.Rmd:
````diff
@@ -0,0 +1,269 @@
+plot(results)
+```
+
+Other multivariate methods are available, such as another type of robust Mahalanobis distance that in this case relies on an orthogonalized Gnanadesikan-Kettenring pairwise estimator [@gnanadesikan1972robust]. These methods are documented and described in the function's [help page](https://easystats.github.io/performance/reference/check_outliers.html).
````
Since I don't have enough expert knowledge in that area, I try to stick to recommendations, and I might be wrong but if one method is considered better in 99% of cases then I think it makes sense to only show the recommended method, even if other methods are available. The question is: are some of those other methods better in > 1% of cases? If so, then, yeah maybe we should. But I don't know. My interpretation of Leys et al., is that they aren't.

Also, I added this sentence just to make it consistent with Dom's addition mentioning other univariate methods. I don't know their use cases though, so I would not be able to elaborate on that. @DominiqueMakowski do you feel knowledgeable enough to elaborate on each method's use case?

strengejacke replied:
This is not critical. If Dom has no other suggestions we can leave it as is

rempsyc replied:
Bump @DominiqueMakowski

rempsyc commented on papers/Mathematics/paper.Rmd:
```diff
@@ -0,0 +1,269 @@
+## Model-Based Outliers
+
+Working with regression models creates the possibility of using model-based SOD methods. These methods rely on the concept of *leverage*, that is, how much influence a given observation can have on the whole model. If few observations have a relatively strong leverage/influence on the model, one can suspect that the model's estimates are biased by these observations, in which case flagging them a ...
+
```
> "can have on the whole model" - what does this mean? Are only estimates affected, or model fit indices as well? So, what is my strongest concern here?

Not sure. That was my paraphrasing of Mattan's

> Something something... what is leverage... why we should care if few observations have (relatively) strong leverage (answer: they are suspect of biasing our estimates!).

I guess I'll use model estimates instead of just model (was trying to simplify for the public but I guess "model" is too vague). I think it would make sense that it affects model fit indices too, but I can't really comment on that though since @mattansb didn't include that in his first sentence. Unless he confirms a better phrasing :)

> "in which case flagging them as outliers could prove helpful." - in which way? what would be the next step where I need this help?

I think this is covered in the next section, "Handling Outliers". But I guess I can a parenthesis, e.g., (see next section, "Handling Outliers").

strengejacke replied:
Sounds good to me

rempsyc commented on papers/Mathematics/paper.Rmd:
```diff
@@ -0,0 +1,269 @@
+Working with regression models creates the possibility of using model-based SOD methods. These methods rely on the concept of *leverage*, that is, how much influence a given observation can have on the whole model. If few observations have a relatively strong leverage/influence on the model, one can suspect that the model's estimates are biased by these observations, in which case flagging them a ...
+
+`check_outliers()` can be applied directly on regression models, specifying `method = "cook"` (or `method = "pareto"` for Bayesian models).^[Our default threshold for the Cook method is defined by `stats::qf(0.5, ncol(x), nrow(x) - ncol(x))`, which again is an approximation of the critical value for _p_ < .001 consistent with the thresholds of our other methods.]
+
```
Ok, I made a summary table:

Table: Summary of Statistical Outlier Detection Methods Recommendations.

|Statistical Test                                           |Diagnosis Method                                   |Recommended Threshold                                   |
|:----------------------------------------------------------|:--------------------------------------------------|:-------------------------------------------------------|
|Supported regression model                                 |Model-based: Cook (or Pareto for Bayesian models)  |qf(0.5, ncol(x), nrow(x) - ncol(x)) (or 0.7 for Pareto) |
|Structural Equation Modeling (or other unsupported model)  |Multivariate: Minimum Covariance Determinant (MCD) |qchisq(p = 1 - 0.001, df = ncol(x))                     |
|Simple test with few variables (t-test, correlation, etc.) |Univariate: robust *z* scores (MAD)                |qnorm(p = 1 - 0.001 / 2), ~ 3.29                        |

rempsyc commented on papers/Mathematics/paper.Rmd:
```diff
@@ -0,0 +1,274 @@
+
+# Handling Outliers
+
+The above section demonstrated how to identify outliers using the `check_outliers()` function in the *{performance}* package. But what should we do with these outliers once identified? Although it is common to automatically discard any observation that has been marked as "an outlier" as if it might infect the rest of the data with its statistical ailment, we believe that the use of SOD methods is ...
```
✔

strengejacke commented on papers/Mathematics/paper.Rmd:
```diff
@@ -0,0 +1,274 @@
+## Winsorization
+
+_Removing_ outliers can in this case be a valid strategy, and ideally one would report results with and without outliers to see the extent of their impact on results. This approach however can reduce statistical power. Therefore, some propose a _recoding_ approach, namely, winsorization: bringing outliers back within acceptable limits [e.g., 3 MADs @tukey1963less]. However, if possible, it is rec ...
+
```
Ah, ok. I think that's fine.

rempsyc commented:
That's it guys, I submitted. And we've done it before the deadline :)

![status](https://user-images.githubusercontent.com/13123390/220211541-43e91a48-28bf-48be-8ee6-d1d9d1d4a564.png)

@strengejacke I know I haven't addressed all your comments yet, but hopefully we can integrate them during the revisions if Dom finds time to answer us 🙃

rempsyc commented:
@strengejacke also regarding the IOAP discount, it seems we have the same, but it's only 10% :(

![discount](https://user-images.githubusercontent.com/13123390/220211776-1c5611d9-05bc-46cc-b9fa-0412a56fbc37.png)

![Hamburg](https://user-images.githubusercontent.com/13123390/220211784-4c07ca8c-df5f-447d-b638-3af1cbf45c0d.png)

Hopefully Dom's editor status can add a tiny 90% to it

rempsyc commented:
We haven't posted it as a preprint though. Should I have done so? It kind of is here on GitHub 😬 But maybe it's not too late to formally do it now if you think that's relevant/useful?

rempsyc commented:
Manuscript update: under review.

![under review](https://user-images.githubusercontent.com/13123390/221382466-0cf0cbad-8921-4429-a7d7-6f2e19dd5fc6.png)

IndrajeetPatil commented:
Now that we have submitted this paper, shouldn't we (squash and) merge the PR?

rempsyc commented:
Reviewer 1:

> In this paper, the authors have shown how to search for outliers using the check_outliers() function of the {performance} package while following current good practices. This contribution will help researchers engage in good search practices while providing an outlier detection experience. They cover univariate, multivariate, and model-based statistical outlier detection methods, their recommended threshold, standard output, and plotting methods. The present paper represents the subject of check the outliers, An accessible introduction to identifying statistical outliers in R with easystats. The study design and methods appropriate for the research question. The results presented clearly and accurately. The authors logically explain the findings.  I highly recommend the publication of this paper in Mathematics Journal.

rempsyc commented:
Reviewer 2: 

> The authors propose the illustration of an R library for detecting outliers. The paper is quite clear, however, the following remarks are provided to improve the article
>
> - [ ] The word accessible in the title should be removed. 
> - [ ] The authors should clarify whether the functions implemented in the library were implemented only for continuous variables or other types of variables are also considered. 
> - [ ] The author's reference to the Normal distribution on page 2, line 31 is unclear. They should clarify or delete this sentence.
> - [ ] The reference to the mathematical score is unclear on page 2, line 36. I think this reference should be removed. 
> - [ ]  A critical review on the analysis of outliers is proposed in this article which must  be cited
>
> Riani, M., & Atkinson, A. C. (2020). Robust regression methods in machine learning. Wiley Interdisciplinary Reviews: Data Mining and Knowledge Discovery, 10(2), e1359. 
> 
> - [ ] Page 3, lines 101-102: It is unclear what the authors mean when referring to BCA, a technique used in non-parametric bootstrapping, but the paper does not mention this.  
> - [ ] It is necessary to change the x-axis labels of all the figures as the numbers overlap; they should be written in a smaller font or otherwise eliminate some of them since they are not readable. I am talking about the figures referring to the histogram of the outliers.
> - [ ] Page seven, line 146, it is better to write association instead of relationship.
> - [ ] Page 7, line 154, the authors should specify what they mean by compatible regression models.
> - [ ] Page 7, code of the first chunk, the authors use the iqr method, but this is not discussed above. They must add details of all the methods they show in the output.
> - [ ] Page 9, line 247, it is difficult to follow the discussion for those unfamiliar with the R datawizard package. What does it do? How does the proposed function tie in with those in the package? 
> - [ ] Page 10, line 272, what does " Registration " mean? 
> - [ ] Check the reference style because it is not uniform

rempsyc commented:
Reviewer 3:

> The paper outlines techniques for processing data with inhomogeneities, such as outliers. However, the style of the article does not correspond to scientific research. The work is educational and auxiliary in nature and is more in line with educational publications, including popular scientific ones.

rempsyc commented:
> shouldn't we (squash and) merge the PR?

I think we can wait until we get the final accepted version, after integrating all the reviewer feedback, etc.

rempsyc commented:
Editor comments:

> - [ ] Less convinced as a scientific research is the major concern. I cannot find a solid contribution for the statistical methodologies, new packages, or sound applications in this study. Almost all the obtained conclusions are known. The readers would be happy to see more convincing results from good real examples from the real world if only the well known statistical methods and packages are used in this study. The tutorial-like results using the examples in R are too weak for an academic paper. It is contributive if the authors can include real examples in different areas to show the power of the used package and present the insight findings, not just show how to use the packages. Then, resubmit the revision as a new paper for review.

rempsyc commented:
**Reviewer 1:** All good.

---

**Reviewer 2:**

> A critical review on the analysis of outliers is proposed in this article which must be cited
Riani, M., & Atkinson, A. C. (2020). Robust regression methods in machine learning. Wiley Interdisciplinary Reviews: Data Mining and Knowledge Discovery, 10(2), e1359.

This reference does not seem to exist, neither by name nor other info. Like it should be volume 10, issue 2 [here](https://wires.onlinelibrary.wiley.com/toc/19424795/2020/10/2), but it's not there. Even the e1359 refers to a completely different article. Anything else we should cite @bwiernik?

> Page 3, lines 101-102: It is unclear what the authors mean when referring to BCA, a technique used in non-parametric bootstrapping, but the paper does not mention this.

We say those methods are documented in the help page, so I think we can ignore this? I don’t think we’re going to explain BCA here…

> It is necessary to change the x-axis labels of all the figures as the numbers overlap; they should be written in a smaller font or otherwise eliminate some of them since they are not readable. I am talking about the figures referring to the histogram of the outliers.

WIP easystats/see#262

> Page 10, line 272, what does " Registration " mean?

Ok I know this was a mathematical journal so reviewers may not be familiar with the word preregistration. But now that we’re submitting to a psych journal, I think that’s pretty ubiquitous.

> Check the reference style because it is not uniform

Looks good to me? 🤔 Not APA but that’s because of the MDPI template.

---

**Editor comment:**

> The readers would be happy to see more convincing results from good real examples from the real world if only the well known statistical methods and packages are used in this study. […] It is contributive if the authors can include real examples in different areas to show the power of the used package and present the insight findings, not just show how to use the packages.

I added the code example of height and weight as we discussed. Think it’s sufficient?

---

**Reviewer 3:** I think that we make at least two original contributions: one by arguing in favour of model-based methods instead of univariate or multivariate. Second, by proposing a novel method: which is the combination of multiple methods. With the added real-life demo of height and weight, I think we got this covered.

rempsyc commented:
I thought we didn't need to merge because we would address the reviewers comments and it would get accepted right away. But now that we need to change journal, I think we should squash and merge this one. And then I will create another PR for _Collabra_ to start with a cleaner workspace. Does that make sense?

bwiernik commented:
> by name nor other info. Like it should be volume 10, issue 2 here, but it's not there. Even the e1359 refers to a completely different article. Anything else we should cite @bwiernik?

Let's just cite the idea of using an outlier-robust distribution like Student T instead of normal, or negative binomial instead of Poisson. For that a good cite is Statistical Rethinking, Chapter 4

rempsyc commented:
> Statistical Rethinking, Chapter 4

Thanks. Couldn't find anything with keywords "robust"  in Chapter 4. However, there is a small mention of it in Chapter 7 (Ulysses' Compass, of the 2020 edition only), I think this must be it:

> One way to both use these extreme observations and reduce their influence is to employ some kind of robust regression . A “robust regression” can mean many different things, but usually it indicates a linear model in which the influence of extreme observations is reduced. A common and useful kind of robust regression is to replace the Gaussian model with a thicker-tailed distribution like Student’s t (or “Student-t”) distribution.

The negative binomial bit is mentioned in Chapter 11. So I think I'll cite the whole book instead of a specific chapter.

bwiernik commented:
There is a section in chapter 4 on using student T distributions

rempsyc commented:
@strengejacke what do you think of the changes? Do you think we can merge this and then I can start a new PR for _Collabra_?

strengejacke commented:
@rempsyc Thanks, your revisions so far look good! Let's merge and open a new PR, right?

rempsyc commented:
Yeah please :)

strengejacke commented:
ok, just open a new PR whenever you want to / can continue with preparing the manuscript.
